From 54428072832d2c7b6f35e9319940fb7103448928 Mon Sep 17 00:00:00 2001
From: Dylan Van Assche <me@dylanvanassche.be>
Date: Sat, 11 Feb 2023 12:46:27 +0100
Subject: [PATCH 08/11] fbd-dev-led-multicolor: Support red, green, blue and
 white LED colors

Provide our own supports_color method to specify which colors are
actually implemented by the driver. RGB LEDs could provide different
colors as well such as orange, purple, etc. but those are not supported
yet.
---
 src/fbd-dev-led-multicolor.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/src/fbd-dev-led-multicolor.c b/src/fbd-dev-led-multicolor.c
index bcb5d17..23fcf54 100644
--- a/src/fbd-dev-led-multicolor.c
+++ b/src/fbd-dev-led-multicolor.c
@@ -144,6 +144,26 @@ fbd_dev_led_multicolor_set_color (FbdDevLed           *led,
 }
 
 
+static gboolean
+fbd_dev_led_multicolor_supports_color (FbdDevLed           *led,
+                                       FbdFeedbackLedColor  color)
+{
+  switch(color) {
+    case FBD_FEEDBACK_LED_COLOR_WHITE:
+      return TRUE;
+    case FBD_FEEDBACK_LED_COLOR_RED:
+      return TRUE;
+    case FBD_FEEDBACK_LED_COLOR_GREEN:
+      return TRUE;
+    case FBD_FEEDBACK_LED_COLOR_BLUE:
+      return TRUE;
+    default:
+      g_warning ("Color unsupported: %d", color);
+      return FALSE;
+  }
+}
+
+
 static void
 fbd_dev_led_multicolor_class_init (FbdDevLedMulticolorClass *klass)
 {
@@ -151,6 +171,7 @@ fbd_dev_led_multicolor_class_init (FbdDevLedMulticolorClass *klass)
 
   fbd_dev_led_class->probe = fbd_dev_led_multicolor_probe;
   fbd_dev_led_class->set_color = fbd_dev_led_multicolor_set_color;
+  fbd_dev_led_class->supports_color = fbd_dev_led_multicolor_supports_color;
 }
 
 
-- 
2.45.1

